rpy2/rinterface_lib/embedded.py

```diff
@@ -141,8 +141,16 @@ def _initr(
         options_c = [ffi.new('char[]', o.encode('ASCII')) for o in _options]
         n_options = len(options_c)
         n_options_c = ffi.cast('int', n_options)
-        status = rlib.Rf_initEmbeddedR(n_options_c,
-                                       options_c)
+
+        # Instead of calling Rf_initEmbeddedR which breaks threaded context
+        # perform the initialization manually to set R_CStackLimit before
+        # calling setup_Rmainloop(), see:
+        # https://github.com/rpy2/rpy2/issues/729
```

Comment:
`Rf_initEmbeddedR` is a rather short function. Wouldn't it make sense to move the setting of interactivity to before `setup_Rmainloop`?

```c
int Rf_initEmbeddedR(int argc, char **argv)
{
    Rf_initialize_R(argc, argv);
    R_Interactive = TRUE;  /* Rf_initialize_R set this based on isatty */
    setup_Rmainloop();
    return(1);
}
```

Reply:
This is interesting because it is only done [on Unix](https://github.com/wch/r-source/blob/d22ee2fc0dc8142b23eed9f46edf76ea9d3ca69a/src/unix/Rembedded.c#L62), but [not on Windows](https://github.com/wch/r-source/blob/58fadda69a5f2db1c1aa1cefe0e32a823cc45a24/src/gnuwin32/embeddedR.c#L128).

I do not understand what effect `R_Interactive` might have on `setup_Rmainloop()`, but since it was not even set some time ago and it is being set just few line further below I thought it would be ok.

Reply:
> This is interesting because it is only done on Unix, but not on Windows.

Yeah, I was trying to cleanup my code base by using `R_initEmbeddedR` (and cut down platform-specific code in rpy2).

> I do not understand what effect R_Interactive might have on setup_Rmainloop(), but since it was not even set some time ago and it is being set just few line further below I thought it would be ok.

Well, it was implicitly set through the call to `R_initEmbeddedR`, wasn't it?

Reply:
Yes it was set implicitly, and then was being overwritten in `rlib.R_Interactive = interactive`. I meant that it was not being set prior to https://github.com/rpy2/rpy2/commit/752f1e8c10896545c35805e2b1b50b0b9afd1d2b (at least for some brief moment, because I see some commented analogue there) - not saying it was good, just that it was like that. Anyways, adding `rlib.R_Interactive = True` just before `rlib.setup_Rmainloop()` should be fine if you wish to keep identical behaviour as previously.

However, I wonder what side effect it may have for a potential user that will invoke `_initr(interactive=False)` - but it might be a separate issue.

Reply:
> Anyways, adding rlib.R_Interactive = True just before rlib.setup_Rmainloop() should be fine if you wish to keep identical behaviour as previously.

That wish comes solely from hoping that `R_initEmbeddedR` does the right thing. That's not certain though.

> However, I wonder what side effect it may have for a potential user that will invoke _initr(interactive=False) - but it might be a separate issue.

It is a good question. Setting `interactive` to `False` is allowed in rpy2 to allow the same behavior as when calling `R CMD BATCH`.